components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/SyncableLoginsStorage.kt

```diff
@@ -194,7 +194,6 @@ class SyncableLoginsStorage(
      */
     @Throws(IdCollisionException::class, InvalidRecordException::class, LoginsStorageException::class)
     override suspend fun add(login: Login): String = withContext(coroutineContext) {
-        check(login.guid == null) { "'guid' for a new login must be `null`" }
```

Comment:
@grigoryk this was breaking the tests because we want to be able to test for matching guids. Think we can get rid of this check or should I make a testing-specific add?

Reply:
Updated